src/quail/sensors/SensorTask.cpp

```diff
@@ -0,0 +1,49 @@
+#include "SensorTask.hpp"
+#include "main.hpp"
+
+StaticEventGroup_t SensorTask::evbuf;
+EventGroupHandle_t SensorTask::evgroup = xEventGroupCreateStatic(&evbuf);
+
+void SensorTask::activity() {
+    while (true) {
+        xEventGroupWaitBits(evgroup, NEED_CONFIG, pdTRUE, pdFALSE, NEVER); // wait for ADC to begin (don't clear bits)
+        configureSensors();
+        while (true) { // read data until ADC restart
+            uint32_t flags = xEventGroupWaitBits(evgroup, NEED_CONFIG | DATA_READY, false, false, NEVER);
+            if (flags & NEED_CONFIG){
+                break;
+            } else {
+                readData();
+            }
+        }
+    }
+}
+
+void SensorTask::configureSensors() {
+    PressureSensor::unconfigure();
+    LoadSensor::unconfigure();
+    ThermalSensor::unconfigure();
```

Comment:
Can we just loop through all sensors and call their .unconfigure() to prevent the need for hard-coding the sensor types? It will call unconfigure() redundantly, but I don't think that does any functional harm as long as its done in a separate loop.

Reply:
Yeah, I agree that makes sense.

Reply:
Yeah but like I mentioned in my review comment - might be that we can't do that, since we cast all sensors to Sensor objects, and the base class doesn't have .unconfigure() as a function (since statics cant be virtual). We could make .unconfigure non-static and give a virtual definition in Sensor, and then have each inheritor's definition of the function call the static _unconfigure() associated with that type of Sensor.